firedrake/utils.py

```diff
@@ -95,3 +95,22 @@ def tuplify(item):
     if not isinstance(item, dict):
         raise ValueError(f"tuplify does not know how to handle objects of type {type(item)}")
     return tuple((k, tuplify(item[k])) for k in sorted(item))
+
+
+def split_by(condition, items):
```

Comment:
This can be achieved more efficiently using `itertools`, see the recipe for `partition`, something like:
```python
import itertools
def itersplit(pred, iterable):
    t1, t2 = itertools.tee(iterable)
    return tuple(filter(pred, t1)), tuple(itertools.filterfalse(pred, t2))
```

Shamelessly stolen from: https://docs.python.org/3/library/itertools.html#itertools-recipes

Reply:
I'd like to claim that my solution works fine and so this isn't a big deal.